client/src/pages/private/ParticipantTable.js

```diff
@@ -173,27 +173,36 @@ export default () => {
   };
 
   const handleFsaFilter = (value) => {
-    setPagination(prev => ({
-      ...prev,
+    if(fsaText && fsaText.trim() === value.trim()){
+      return
+    }
+    setPagination(oldPagination => ({
+      ...oldPagination,
       currentPage: 0,
     }));
-    setFsaFilter(value);
+    setFsaFilter(value.trim());
   };
 
   const handleLastNameFilter = (value) => {
-    setPagination(prev => ({
-      ...prev,
+    if(lastNameText && lastNameText.trim() === value.trim()){
```

Comment:
```suggestion
    if(lastNameText?.trim() === value?.trim()){
```

Little bit less wordy option